kolibri/core/content/views.py

```diff
@@ -164,20 +195,15 @@ def get(self, request, zipped_filename, embedded_filepath):
 
         with zipfile.ZipFile(zipped_path) as zf:
 
+            # handle H5P files
+            if zipped_path.endswith('h5p') and (not embedded_filepath or embedded_filepath.startswith("dist/")):
+                return self.get_h5p(request, zf, embedded_filepath)
+
             # if no path, or a directory, is being referenced, look for an index.html file
             if not embedded_filepath or embedded_filepath.endswith("/"):
                 embedded_filepath += "index.html"
 
-            # get the details about the embedded file, and ensure it exists
```

Comment:
This is intentionally invoked before any Hashi processing to return a 404 for non-existent html files, and not bother returning the hashi template. This is why you are getting a test failure, as this case is specifically tested for.